docs/madr/decisions/011-mesh-traffic-route.md

```diff
@@ -0,0 +1,902 @@
+balancing configuration.
+
+The main motivator for this is that load balancing is done on an Envoy cluster.
+Including load balancing options on a match would force creation of a new cluster for that
+set of `backendRefs` which has negative consequences including but not limited to
+metrics handling and a potentially large number of clusters.
+
+#### Merging behavior
+
+With `targetRef` resources, the ultimate configuration of a data plane proxy
+is always a result of all policies targeting that proxy depending on their
+specificity.
+
+> :question: Is the current merge behavior what we need for routing policy? What surprises are there? See the next sections
+
+This proposal addresses one issue with a "naive" application of `targetRef`
+structure to a routing policy.
+
+##### Surprising behavior of top level `default`
+
+Let's look at two basic theoretical route policies where we include `default` at
+the top level of a `to` rule:
+
+```yaml
+metadata:
+ name: owner
+spec:
+ targetRef:
+  kind: Mesh
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    default:
+     rules:
+      - matches:
+         path:
+          prefix: /v1
+        backendRefs:
+         - weight: 100
+           kind: MeshServiceSubset
+           name: backend
+           tags:
+            version: v1
+      - matches:
+         path:
+          prefix: /v2
+        backendRefs:
+         - weight: 100
+           kind: MeshServiceSubset
+           name: backend
+           tags:
+            version: v2
+---
+metadata:
+ name: consumer
+spec:
+ targetRef:
+  kind: MeshService
+  name: frontend
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    default:
+     rules:
+      - matches:
+         path:
+          prefix: /v2
+        filters:
+         - requestHeaderModifier:
+            add:
+             - name: env
+               value: dev
+        backendRefs:
+         - weight: 100
+           kind: MeshServiceSubset
+           name: backend
+           tags:
+            version: v2
+```
+
+The rules for merging a field in a `targetRef` policy is that more specific
+routes replace the value from less specific routes. So the final value of `rules`
+is the value from the `consumer` route:
+
+```yaml
+spec:
+ targetRef:
+  kind: MeshService
+  name: frontend
+ to:
+  - targetRef:
+     kind: MeshService
+     name: backend
+    rules:
+     - matches:
+        path:
+         prefix: /v2
+       filters:
+        - requestHeaderModifier:
+           add:
+            - name: env
+              value: dev
+       backendRefs:
+        - weight: 100
+          kind: MeshServiceSubset
+          name: backend
+          tags:
+           version: v2
+```
+
+This is likely surprising, especially because `consumer` didn't specify anything
+at all for `/v1`.
+
+##### Solution
+
+This MADR proposes to distribute `default` down into individual `rules` and merge
```

Comment:
What if `consumer` wants to override the entire list of rules from the owner?

Reply:
They can't. There's trade-offs for every option.

Reply:
Added. I think we can write the types such that we can override using an explicit `filters: []`